std/algorithm/sorting.d

```diff
@@ -1698,17 +1698,30 @@ private void shortSort(alias less, Range)(Range r)
                 auto t = r[0]; if (pred(t, r[0])) r[0] = r[0];
             }))) // Can we afford to temporarily invalidate the array?
         {
+            import core.lifetime : move;
+
             size_t j = i + 1;
-            auto temp = r[i];
+            static if (hasLvalueElements!Range)
+                auto temp = move(r[i]);
```

Comment:
So, for a range with lvalue elements, all the `move`s inside the following `if` can all now be `moveEmplace`s, to avoid unnecessary dtor calls ;)

Reply:
Fixed